crates/pathfinder/src/bin/pathfinder.rs

```diff
@@ -32,25 +32,37 @@ async fn main() -> anyhow::Result<()> {
     let eth_transport =
         HttpTransport::from_config(config.ethereum).context("Creating Ethereum transport")?;
 
-    let network_chain = eth_transport
+    let ethereum_chain = eth_transport
         .chain()
         .await
         .context("Determining Ethereum chain")?;
 
-    let database_path = config.data_directory.join(match network_chain {
+    let database_path = config.data_directory.join(match ethereum_chain {
         ethereum::Chain::Mainnet => "mainnet.sqlite",
         ethereum::Chain::Goerli => "goerli.sqlite",
     });
     let storage = Storage::migrate(database_path.clone()).unwrap();
     info!(location=?database_path, "Database migrated.");
 
-    let sequencer = sequencer::Client::new(network_chain).unwrap();
+    let sequencer = match config.sequencer_url {
+        Some(url) => {
+            info!(?url, "Using custom Sequencer address");
+            let client = sequencer::Client::with_url(url).unwrap();
+            let sequencer_chain = client.chain().await.unwrap();
```

Comment:
This departs a bit from our current sequencer usage in that `sync` automatically restarts on failure. Whereas this will just end the node on a terminal failure, like decoding error. But maybe that's okay?